tests/ci/integration/libwebsockets_patch/libwebsocket_master.patch

```diff
@@ -0,0 +1,420 @@
++#if defined(LWS_WITH_BORINGSSL) || defined(LWS_WITH_AWSLC)
+ 				(uint32_t)
+ #else
+ #if (OPENSSL_VERSION_NUMBER >= 0x10003000l) && \
+@@ -945,7 +945,7 @@ lws_tls_client_create_vhost_context(struct lws_vhost *vh,
+ #if (OPENSSL_VERSION_NUMBER >= 0x009080df) && !defined(USE_WOLFSSL)
+
+ 	/* SSL_clear_options introduced in 0.9.8m */
+-#if defined(LWS_WITH_BORINGSSL)
++#if defined(LWS_WITH_BORINGSSL) || defined(LWS_WITH_AWSLC)
+                 uint32_t
+ #else
+ #if (OPENSSL_VERSION_NUMBER >= 0x10003000l) && \
+@@ -957,7 +957,7 @@ lws_tls_client_create_vhost_context(struct lws_vhost *vh,
+ #endif
+
+ 			ssl_client_options_clear_value =
+-#if defined(LWS_WITH_BORINGSSL)
++#if defined(LWS_WITH_BORINGSSL) || defined(LWS_WITH_AWSLC)
+ 				(uint32_t)
+ #else
+ #if (OPENSSL_VERSION_NUMBER >= 0x10003000l) && \
+@@ -1098,7 +1098,7 @@ lws_tls_client_create_vhost_context(struct lws_vhost *vh,
+ 		}
+
+ 		n = SSL_CTX_use_certificate_ASN1(vh->tls.ssl_client_ctx,
+-#if defined(LWS_WITH_BORINGSSL)
++#if defined(LWS_WITH_BORINGSSL) || defined(LWS_WITH_AWSLC)
+ 				(size_t)
+ #else
+ 				(int)
+@@ -1149,7 +1149,7 @@ lws_tls_client_create_vhost_context(struct lws_vhost *vh,
+ 		}
+
+ 		n = SSL_CTX_use_PrivateKey_ASN1(EVP_PKEY_RSA, vh->tls.ssl_client_ctx, p,
+-#if defined(LWS_WITH_BORINGSSL)
++#if defined(LWS_WITH_BORINGSSL) || defined(LWS_WITH_AWSLC)
+ 				(size_t)
+ #else
+ 				(long)(lws_intptr_t)
+@@ -1158,7 +1158,7 @@ lws_tls_client_create_vhost_context(struct lws_vhost *vh,
+ 		if (n != 1)
+ 			n = SSL_CTX_use_PrivateKey_ASN1(EVP_PKEY_EC,
+ 							vh->tls.ssl_client_ctx, p,
+-#if defined(LWS_WITH_BORINGSSL)
++#if defined(LWS_WITH_BORINGSSL) || defined(LWS_WITH_AWSLC)
+ 				(size_t)
+ #else
+ 				(long)(lws_intptr_t)
+diff --git a/lib/tls/openssl/openssl-server.c b/lib/tls/openssl/openssl-server.c
+index a3c8d120..8c768b21 100644
+--- a/lib/tls/openssl/openssl-server.c
++++ b/lib/tls/openssl/openssl-server.c
+@@ -159,7 +159,7 @@ lws_tls_server_certs_load(struct lws_vhost *vhost, struct lws *wsi,
+     ((OPENSSL_VERSION_NUMBER < 0x30000000l) || \
+      defined(LWS_SUPPRESS_DEPRECATED_API_WARNINGS))
+ 	const char *ecdh_curve = "prime256v1";
+-#if !defined(LWS_WITH_BORINGSSL) && defined(LWS_HAVE_SSL_EXTRA_CHAIN_CERTS)
++#if !defined(LWS_WITH_BORINGSSL) && !defined(LWS_WITH_AWSLC) && defined(LWS_HAVE_SSL_EXTRA_CHAIN_CERTS)
+ 	STACK_OF(X509) *extra_certs = NULL;
+ #endif
+ 	EC_KEY *ecdh, *EC_key = NULL;
+@@ -216,7 +216,7 @@ lws_tls_server_certs_load(struct lws_vhost *vhost, struct lws *wsi,
+ 			error = ERR_get_error();
+
+ 			s = ERR_error_string(
+-#if defined(LWS_WITH_BORINGSSL)
++#if defined(LWS_WITH_BORINGSSL) || defined(LWS_WITH_AWSLC)
+ 				(uint32_t)
+ #endif
+ 					error,
+@@ -238,7 +238,7 @@ lws_tls_server_certs_load(struct lws_vhost *vhost, struct lws *wsi,
+ 				const char *s;
+ 				error = ERR_get_error();
+ 				s = ERR_error_string(
+-	#if defined(LWS_WITH_BORINGSSL)
++	#if defined(LWS_WITH_BORINGSSL) || defined(LWS_WITH_AWSLC)
```

Comment:
It will be worth while to try and simplify this upstream for future releases.  This #ifdef could simply turn into something like the below, which will allow us to remove a lot of these #ifdefs in libwebsockets.  What we have here is more than fine, it follows the pattern the project set, but I think we should minimize the total number of #ifdefs.  

`ERR_error_string(ERR_get_error(), (char*)vhost->context->pt[0].serv_buf;)`